### PR TITLE
Looking to merge AddProductPage first

### PR DESCRIPTION
**Pull request by ricky-taing**
…oved AddProductPage to new branch.

### CODE REVIEW

File: src/components/Products/CheckoutModal.js [hunk 1]
```diff
@@ -8,64 +8,53 @@ class CheckoutModal extends React.Component {
...
                 timeStamp: Date.now(),
                 totalPrice: this.state.totalPrice,
                 userID:this.props.propId
             }
-            console.log("purchased cart"+purchasedCart)
+            console.log(purchasedCart)
             //add to purchase
             let userPurchases= (await Fire.database().ref(`user/${this.props.propId}`).once('value')).val().purchase;
-            let purchaseRef = Fire.database().ref(`purchase`).push(purchasedCart)
+            let purchaseRef = Fire.database().ref(`user/${this.props.propId}/purchase`).push(purchasedCart)
```
Comment (Anamm1997) [inline src/components/Products/CheckoutModal.js:54]: This is wrong. use 
let purchaseRef = Fire.database().ref(`purchase`).push(purchasedCart)

File: src/components/Products/CheckoutModal.js [hunk 2]
```diff
@@ -8,64 +8,53 @@ class CheckoutModal extends React.Component {
...
+                cart: this.state.currentCart,
                 timeStamp: Date.now(),
                 totalPrice: this.state.totalPrice,
                 userID:this.props.propId
             }
-            console.log("purchased cart"+purchasedCart)
+            console.log(purchasedCart)
             //add to purchase
             let userPurchases= (await Fire.database().ref(`user/${this.props.propId}`).once('value')).val().purchase;
-            let purchaseRef = Fire.database().ref(`purchase`).push(purchasedCart)
```
Comment (Anamm1997) [inline src/components/Products/CheckoutModal.js]: Revert to this

File: src/components/Products/CheckoutModal.js [hunk 3]
```diff
@@ -8,64 +8,53 @@ class CheckoutModal extends React.Component {
...
+            let purchaseRef = Fire.database().ref(`user/${this.props.propId}/purchase`).push(purchasedCart)
             if(!userPurchases) {userPurchases = []}
             userPurchases.push(purchaseRef.key);
-
+
```
Comment (Anamm1997) [inline src/components/Products/CheckoutModal.js:57]: add

Fire.database().ref(`user/${this.props.propId}`).update(userPurchases)

Comment (LindaRZhang) [reply src/components/Products/CheckoutModal.js:57]: Got permission denied

**Pull request merged**